src/rabbit_channel.erl

```diff
@@ -789,9 +792,16 @@ check_internal_exchange(#exchange{name = Name, internal = true}) ->
 check_internal_exchange(_) ->
     ok.
 
-check_topic_authorisation(#exchange{name = Name, type = topic}, #ch{user = User}, RoutingKey, Permission) ->
+check_topic_authorisation(#exchange{name = Name = #resource{virtual_host = VHost}, type = topic},
+                          #ch{user = User = #user{username = Username}, conn_pid = ConnPid},
+                          RoutingKey,
+                          Permission) ->
     Resource = Name#resource{kind = topic},
-    Context = #{routing_key => RoutingKey},
+    AmqpConnInfo = gen_server:call(ConnPid, {info, [amqp_params]}, ?TIMEOUT_60s),
```

Comment:
@lukebakken I don't think so, that PR simply copies what https://github.com/rabbitmq/rabbitmq-common/blob/ceec9bd6109eebbeddc9d4706ce1d71be815d7c2/src/rabbit_channel.erl#L415 does.

We can extract this into a function (in this repo). It's a good point that while this call is to a node-local process, it is on the hot path.